On 20081011, Khem RAJ writes:
 You have to bar gcc trying to build unwinding at stage 1. Try the
 attached patch and see if it helps.

 gcc/config/sh/linux-unwind.h |    3 +++
 1 file changed, 3 insertions(+)

Index: gcc-4_2-branch/gcc/config/sh/linux-unwind.h
===================================================================
--- gcc-4_2-branch.orig/gcc/config/sh/linux-unwind.h	2008-07-01 17:24:40.000000000 -0700
+++ gcc-4_2-branch/gcc/config/sh/linux-unwind.h	2008-07-01 17:25:47.000000000 -0700
@@ -26,6 +26,8 @@ along with this program; see the file CO
 the Free Software Foundation, 51 Franklin Street, Fifth Floor,
 Boston, MA 02110-1301, USA.  */
 
+#ifndef inhibit_libc
+
 /* Do code reading to identify a signal frame, and set the frame
    state data appropriately.  See unwind-dw2.c for the structs.  */
 
@@ -251,3 +253,4 @@ sh_fallback_frame_state (struct _Unwind_
   return _URC_NO_REASON;
 }
 #endif /* defined (__SH5__) */
+#endif /* ifndef inhibit_libc */

